File: examples/typescript/list-invoices.ts
/**
 * List Invoices Example
 *
 * List your sent invoices with pagination, or iterate through all of them.
 * Note: invoices.list() returns outbound submissions only (invoices you sent).
 */

import { Peppol } from "@getpeppr/sdk";

const peppol = new Peppol({ apiKey: "sk_sandbox_..." });

// List your invoices (paginated)
const page = await peppol.invoices.list({ limit: 10, offset: 0 });

console.log(`Total invoices: ${page.meta.totalCount} (showing ${page.data.length})`);

for (const invoice of page.data) {
  console.log(`  ${invoice.id}: ${invoice.number} — ${invoice.status}`);
}

// Paginate through all invoices
for await (const invoice of peppol.invoices.listAll()) {
  console.log(`${invoice.number}: ${invoice.status}`);
}

File: examples/python/list_invoices.py
"""List your sent invoices via the getpeppr API.

Note: GET /v1/invoices returns outbound submissions only (invoices you sent).

The response shape is {"invoices": [...], "meta": {...}}. Items carry
"invoiceNumber" (not "number"); "meta" uses snake_case ("total_count").
The TypeScript example reads page.data / invoice.number because the SDK maps
those names — raw HTTP, as used here, does not.
"""

import requests

BASE_URL = "https://api.getpeppr.dev"
API_KEY = "sk_sandbox_abc123..."

response = requests.get(
    f"{BASE_URL}/v1/invoices",
    headers={"Authorization": f"Bearer {API_KEY}"},
    params={"limit": 10, "offset": 0},
    timeout=30,
)

response.raise_for_status()
result = response.json()

print(f"Total invoices: {result['meta']['total_count']} (showing {len(result['invoices'])})")

for invoice in result["invoices"]:
    print(f"  {invoice['id']}: {invoice['invoiceNumber']} — {invoice['status']}")
